dask/array/core.py

```diff
@@ -2791,7 +2831,7 @@ def concatenate(seq, axis=0, allow_unknown_chunksizes=False):
     dsk = dict(zip(keys, values))
     graph = HighLevelGraph.from_collections(name, dsk, dependencies=seq)
 
-    return Array(graph, name, chunks, dtype=dt)
+    return Array(graph, name, chunks, meta=meta.astype(dt))
```

Comment:
I would expect the `dt` logic to no longer be necessary now that we've called concatenate on all of the meta objects.  I would hope that as we add meta code we can hopefully remove dtype code.  In particular 2815:2820 ?

Reply:
Yes, I got rid of them. The reason I hadn't done that was a test would fail. Please check https://github.com/dask/dask/pull/4543/commits/f5281ee5e40ff0d525ae389eff54f4da54029702 to see what I did now to fix the test, and the reasoning behind my solution is that in both cases dtypes get promoted to `float64`:

```python
In [1]: import numpy as np

In [2]: np.promote_types(np.dtype('>f8'), np.dtype('>f8'))
Out[2]: dtype('float64')

In [3]: np.promote_types(np.dtype('<f4'), np.dtype('<f8'))
Out[3]: dtype('float64')
```

Reply:
Actually, I've removed too much. There's still the need to properly adjust the actual data dtype, only `_meta` doesn't need that. I added back a part of it in https://github.com/dask/dask/pull/4543/commits/2a62263cbd97dfe7b36266205d6a748dd7362755.